hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/SqlPrunedMemberLeftTest.java

```diff
@@ -0,0 +1,149 @@
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.jet.core.JetTestSupport;
+import com.hazelcast.jet.datamodel.Tuple2;
+import com.hazelcast.jet.sql.impl.misc.Pojo;
+import com.hazelcast.map.IMap;
+import com.hazelcast.spi.impl.NodeEngine;
+import com.hazelcast.sql.SqlRow;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.hazelcast.jet.sql.SqlTestSupport.createMapping;
+import static com.hazelcast.jet.sql.impl.SqlEndToEndTestSupport.calculateExpectedPartitions;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNotSame;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class SqlPrunedMemberLeftTest extends JetTestSupport {
+    private static final int MAP_ENTRIES = 1_000;
+    private static final String MAP_NAME = "map";
+
+    private NodeEngine nodeEngine1;
+
+    private HazelcastInstance hz1;
+    private HazelcastInstance hz2;
+    private HazelcastInstance hz3;
+    private HazelcastInstance hz4;
+    private HazelcastInstance hz5;
+
+    private Map<Address, HazelcastInstance> addrToInstanceMap;
+
+    @Before
+    public void setUp() throws Exception {
+        hz1 = createHazelcastInstance(regularInstanceConfig());
+        nodeEngine1 = getNodeEngineImpl(hz1);
+        hz2 = createHazelcastInstance(regularInstanceConfig());
+        hz3 = createHazelcastInstance(regularInstanceConfig());
+        hz4 = createHazelcastInstance(regularInstanceConfig());
+        hz5 = createHazelcastInstance(regularInstanceConfig());
+
+        addrToInstanceMap = new HashMap<>();
+        addrToInstanceMap.put(hz1.getCluster().getLocalMember().getAddress(), hz1);
+        addrToInstanceMap.put(hz2.getCluster().getLocalMember().getAddress(), hz2);
+        addrToInstanceMap.put(hz3.getCluster().getLocalMember().getAddress(), hz3);
+        addrToInstanceMap.put(hz4.getCluster().getLocalMember().getAddress(), hz4);
+        addrToInstanceMap.put(hz5.getCluster().getLocalMember().getAddress(), hz5);
+
+        prepareMap();
+        createMapping(hz5, MAP_NAME, Pojo.class, Integer.class);
+    }
+
+    @Test
+    // https://hazelcast.atlassian.net/browse/HZ-2992
+    public void test() {
+        // Given
+        final int c = 2; // constant, assuming hz3 will be coordinator.
```

Comment:
I do not see any relationship between `c = 2` and `hz3 will be coordinator`

Reply:
that is how hash function distribution works: 
c = 0 -> hz1
c = 1 -> hz2
c = 2 -> hz3